lib/matplotlib/units.py

```diff
@@ -157,11 +157,15 @@ def get_converter(self, x):
                     converter = self.get_converter(next_item)
                 return converter
 
-        if converter is None and iterable(x) and (len(x) > 0):
-            thisx = safe_first_element(x)
-            if classx and classx != getattr(thisx, '__class__', None):
-                converter = self.get_converter(thisx)
-                return converter
+        if converter is None:
+            try:
+                thisx = safe_first_element(x)
+            except (TypeError, StopIteration):
+                pass
+            else:
+                if classx and classx != getattr(thisx, '__class__', None):
+                    converter = self.get_converter(thisx)
+                    return converter
```

Comment:
I tried to keep the logical flow of the previous implementation as much as possible -- I think the whole thing probably needs some refactor anyways, so there isn't much in favor of a piecemeal approach.